test/utils/reflow/respace_test.py

```diff
@@ -0,0 +1,107 @@
+"""Tests for respacing methods.
+
+These are mostly on the ReflowPoint class.
+"""
+
+import logging
+import pytest
+
+from sqlfluff.core import Linter
+from sqlfluff.utils.reflow.elements import ReflowPoint
+
+from sqlfluff.utils.reflow.sequence import ReflowSequence
+
+
+def parse_ansi_string(sql, config):
+    """Parse an ansi sql string for testing."""
+    linter = Linter(config=config)
+    return linter.parse_string(sql).tree
+
+
+@pytest.mark.parametrize(
+    "raw_sql_in,kwargs,raw_sql_out",
+    [
+        # Basic cases
+        ("select 1+2", {}, "select 1 + 2"),
+        ("select    1   +   2    ", {}, "select 1 + 2"),
+        # Check newline handling
+        ("select\n    1   +   2", {}, "select\n    1 + 2"),
+        ("select\n  1   +   2", {}, "select\n  1 + 2"),
+        ("select\n  1   +   2", {"strip_newlines": True}, "select 1 + 2"),
+        # Check filtering
+        ("select  \n  1   +   2 \n ", {}, "select\n  1 + 2\n"),
+        ("select  \n  1   +   2 \n ", {"filter": "all"}, "select\n  1 + 2\n"),
+        ("select  \n  1   +   2 \n ", {"filter": "inline"}, "select  \n  1 + 2 \n "),
+        ("select  \n  1   +   2 \n ", {"filter": "newline"}, "select\n  1   +   2\n"),
+    ],
+)
+def test_reflow__sequence_respace(
+    raw_sql_in, kwargs, raw_sql_out, default_config, caplog
+):
+    """Test the ReflowSequence.respace() method directly."""
+    root = parse_ansi_string(raw_sql_in, default_config)
+    seq = ReflowSequence.from_raw_segments(
+        root.raw_segments, root, config=default_config
+    )
+
+    with caplog.at_level(logging.DEBUG, logger="sqlfluff.rules.reflow"):
+        new_seq = seq.respace(**kwargs)
+
+    assert new_seq.get_raw() == raw_sql_out
+
+
+@pytest.mark.parametrize(
+    "raw_sql_in,point_idx,kwargs,raw_point_sql_out,fixes_out",
+    [
+        # Basic cases
+        ("select    1", 1, {}, " ", {("replace", "    ")}),
```

Comment:
For `replace`, should there be another value? (I.e. we're replacing X with Y.)

Reply:
Yes there is, but that's kind of tested using the `raw_point_sql_out` variable. The `fixes_out` option is much more just the check that the right kind and volume of fixes are generated.